src/PresentationalComponents/RulesTable/RulesTable.js

```diff
@@ -64,21 +64,23 @@ const RulesTable = ({ rules, filters, rulesFetchStatus, setFilters, fetchRules,
         setOffset(0);
     }, [setSort, setSortBy, setOffset]);
 
-    const onSetPage = (_event, pageNumber) => {
+    const onSetPage = (pageNumber) => {
         const newOffset = pageNumber * limit - limit;
         setOffset(newOffset);
     };
 
-    const onPerPageSelect = (_event, limit) => {
-        setLimit(limit);
-    };
-
     const toggleRulesWithHits = (impacting) => {
         setFilters({ ...filters, impacting });
         setImpacting(impacting);
         setOffset(0);
     };
 
+    const toggleRulesDisabled = (reports_shown) => {
+        setFilters({ ...filters, reports_shown });
+        setReportsShown(reports_shown);
+        setOffset(0);
```

Comment:
I would consider using `useReducer` for this component. You are doing batch updates which are not playing well with `useState`. This callback will do one re-render per setState call. Using `useReducer` you can bundle this to one dispatch which do only one state update and only one render. Probably not in this PR but something that might be good to think about.

Reply:
oooo https://reactjs.org/docs/hooks-reference.html#usereducer thanks for the tip @Hyperkid123 🤗